### PR TITLE
added version check for CLI binary

### PR DESCRIPTION
**Pull request by jasmingacic**
# Summary of this proposal

Checking CLI version before running subcommands .

```
# ./calyptia get core_instances 
Warning: This version 1.0.0 of Calyptia cli is outdated. The latest version available is 1.12.1
NAME        VERSION ENVIRONMENT PIPELINES TAGS STATUS      AGE
testsecrets v2.1.1  default     2              unreachable 1 hour
```
disabling the check
```
# ./calyptia get core_instances --disable-version-check
NAME        VERSION ENVIRONMENT PIPELINES TAGS STATUS      AGE
testsecrets v2.1.1  default     2              unreachable 1 hour
```
Check is disabled by checking **isatty** so when piping the output won't be messed up

```
# ./calyptia get core_instances -o yaml | yq
- id: 4d74f2cb-a3ba-4cec-8d16-a9c5f8b95ffb
  token: ....
```

## Asana task(s) link.

https://app.asana.com/0/1204316201844829/1206101751551634

## Notes for the reviewer

Ideally we should be doing checks for image update but I would rather rework update to do reinstall of the operator to ensure that the whole package is indeed picking up the latest version.

### CODE REVIEW

File: cmd/root.go
Comment (patrick-stephens) [inline cmd/root.go:105]: Let's add an env var as well so we don't have to keep adding the flag or set up an alias.

File: cmd/version/version.go
Comment (patrick-stephens) [inline cmd/version/version.go:9]: How come you changed this?

File: cmd/version/githubclient.go
Comment (patrick-stephens) [inline cmd/version/githubclient.go:1]: Why not use an existing library like https://github.com/google/go-github?
Feels like we're reinventing stuff and have to then maintain it.

File: cmd/root.go
Comment (patrick-stephens) [inline cmd/root.go:127]: Rate limiting will presumably just drop out silently along with any other http errors?

File: cmd/version/version.go
Comment (jasmingacic) [reply cmd/version/version.go:9]: forgot to return it while testing

File: cmd/version/githubclient.go
Comment (jasmingacic) [reply cmd/version/githubclient.go:1]: go-github has lot more than what we need. So I just took that one part without importing another library.

File: cmd/root.go
Comment (jasmingacic) [reply cmd/root.go:127]: If it kicks in it will be ignored

File: cmd/root.go
Comment (jasmingacic) [reply cmd/root.go:108]: done

Comment (niedbalski) [reply cmd/root.go:108]: +1 to this

Comment (jasmingacic) [reply cmd/root.go:108]: https://github.com/calyptia/cli/pull/714/files#diff-ab967ab1a2f3a1b769106eeb7bfe892ef0e81d1d27811fa15be08e6749feee1fR31

File: cmd/root.go
Comment (nicolasparada) [inline cmd/root.go:125]: There are two `if`s on this function that should be inverted to reduce indentation.

File: cmd/root.go
Comment (nicolasparada) [inline cmd/root.go:19]: `utils` is a duplicate import of `cliversion` (An also a terrible name).

File: cmd/version/version.go
Comment (nicolasparada) [inline cmd/version/version.go:9]:
```suggestion
	Version            = "dev" // To be injected at build time:  -ldflags="-X 'github.com/calyptia/cli/cmd/version.Version=xxx'"
```

File: cmd/version/githubclient.go
Comment (nicolasparada) [inline cmd/version/githubclient.go:89]: missing error handling

File: cmd/version/githubclient.go
Comment (nicolasparada) [inline cmd/version/githubclient.go:92]: missing error handling

File: cmd/version/githubclient.go
Comment (nicolasparada) [inline cmd/version/githubclient.go:88]: missing error handling

File: cmd/version/githubclient.go
Comment (nicolasparada) [inline cmd/version/githubclient.go:112]: missing error handling

File: cmd/version/githubclient.go
Comment (nicolasparada) [inline cmd/version/githubclient.go:118]: I don't get this API... This returns an `error` but also a response...
`*ErrorResponse` should instead implement `error` and return just that. Not both values.

File: cmd/version/githubclient.go
Comment (nicolasparada) [inline cmd/version/githubclient.go:61]:
```suggestion
type GitHubError struct {
```

File: cmd/version/githubclient.go
Comment (nicolasparada) [inline cmd/version/githubclient.go:14]:
```suggestion
	githubBaseURL = "https://api.github.com/repos/calyptia/"
```

File: cmd/version/githubclient.go
Comment (nicolasparada) [inline cmd/version/githubclient.go:134]: All these types are for the github API client, but are on the `version` package...
Maybe rename them to include the work "github" or move the entire github client to a new package.

File: cmd/version/githubclient.go
Comment (nicolasparada) [inline cmd/version/githubclient.go:174]: Only `tag_name` is used really. So there is no need to unmarshall and keep track of all the fields.

File: cmd/version/githubclient.go
Comment (nicolasparada) [inline cmd/version/githubclient.go:28]:
```suggestion
func (c *GithubClient) GetLatestTag(repo string) (string, error) {
```

**Review by nicolasparada — CHANGES_REQUESTED** [review]
Left a few comments.

File: cmd/version/githubclient.go
Comment (nicolasparada) [inline cmd/version/githubclient.go:37]: I don't think constructors should error.

File: cmd/version/githubclient.go
Comment (jasmingacic) [reply cmd/version/githubclient.go:89]: it is intentional

File: cmd/version/githubclient.go
Comment (jasmingacic) [reply cmd/version/githubclient.go:118]: there is no need for this actually

Comment (jasmingacic) [reply cmd/version/githubclient.go:118]: all errors are ignored here anyways

File: cmd/version/githubclient.go
Comment (jasmingacic) [reply cmd/version/githubclient.go]: done

**Review by niedbalski — APPROVED** [review]
LGTM

**Review by patrick-stephens — APPROVED** [review]

**Comment by patrick-stephens** [comment]
@nicolasparada please approve

**Comment by niedbalski** [comment]
> I'll approve for now but I don't like this approach of ignoring all the errors and just continuing... It's not the way of how Go works.

@nicolasparada let's don't merge this if we have no agreement. Can you expand on your concerns here?

**Comment by jasmingacic** [comment]
This has been sitting here since December 
I'm closing it as it has become stale

**Pull request closed without merging**
